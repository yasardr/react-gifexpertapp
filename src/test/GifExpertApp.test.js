import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';

describe('Probando GifExpertApp', () => { 
    test('debe de mostrar GifExpertApp correctamente', () => { 
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de emostrar una lista de categorias', () => { 
        const categories = ['One Piece', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
});