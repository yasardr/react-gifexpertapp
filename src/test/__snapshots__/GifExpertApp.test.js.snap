// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifExpertApp debe de emostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Probando GifExpertApp debe de mostrar GifExpertApp correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
